### ISSUE TITLE
[5.1] Using pluck() on Illuminate\Database\Eloquent\Collection returns an Illuminate\Database\Eloquent\Collection which results in getKey errors when using further Collection methods.

### ISSUE THREAD

**Issue by TommyC81** [laravel/framework#9163]
When using the pluck() function on an Eloquent hasMany relationship an Illuminate\Database\Eloquent\Collection (rather than Illuminate\Support\Collection) is returned. This causes issues when further functions is used on the Collection:

```
$questionTypes = $exam->examQuestions->pluck('type');
$uniqueTypesOfQuestionsUsed = $questionTypes->unique()->count();
```

Error:
```
FatalErrorException in Collection.php line 248:
Call to a member function getKey() on integer
```

The following Illuminate\Database\Eloquent\Collection -> array -> Illuminate\Support\Collection conversion is required to make it work as expected:
```
$questionTypes = collect($exam->examQuestions->pluck('type')->all());
$uniqueTypesOfQuestionsUsed = $questionTypes->unique()->count();
```
Intentional?

**Comment by Sladewill** [comment]
Pretty much all Collection methods will return a new instance of Collection now use all to return an array if that's expected behavior.

**Comment by GrahamCampbell** [comment]
Yeh, this stuff was changed if you're upgrading to Laravel 5.1. :)

**Comment by TommyC81** [comment]
Hmm, not sure if you actually read my description?

The problem is not that a collection is returned but the seemingly unintentional problem of it being an "EloquentCollection" which assumes a getKey function exists for each item (thus causing an error as described above).

It's great that collections are returned, but having to convert a collection to an array and then back to a collection doesn't seem like the Laravel way... 

Makes sense?

**Comment by TommyC81** [comment]
@GrahamCampbell consider reopening?

**Comment by GrahamCampbell** [comment]
Sorry, I just read @Sladewill's comment there.

**Comment by TommyC81** [comment]
@Sladewill just want to check if you actually read the issue properly so I'm not missing something.

**Comment by Sladewill** [comment]
Sorry only skimmed the issue, misread what was the issue.

**Comment by Sladewill** [comment]
Looking at the class anything returning a new Collection from the database class is going to have this issue as its calling a new static on Illuminate\Database\Eloquent\Collection even if it isnt a model anymore, maybe replicate the command in the Eloquent version and return new parent(...)

**Comment by rkgrep** [comment]
> maybe replicate the command in the Eloquent version and return new parent(...)

I think it would be a best solution.

**Comment by GrahamCampbell** [comment]
We're open to PRs.

**Comment by ibrasho** [comment]
5.1, 5.2 or master?

**Comment by GrahamCampbell** [comment]
5.2 is master. I'd say 5.1 is good, though Taylor might ask you to retarget.

**Comment by tonglil** [comment]
You could use "->toBase()": `$questionTypes = $exam->examQuestions->pluck('type')->toBase()`.

**Comment by TommyC81** [comment]
@tonglil Thanks buddy, can do that, not much of a difference from the workaround though. Believe the issue was an unintentional error that snuck into an otherwise awesome update.

**Comment by AaronJan** [comment]
So this will not be fixed in `5.1`, then how about add some notes for `5.1 docs`?

because I just had this problem today, I believe many others will too.
